src/components/Notification/InactivityMessage.jsx

```diff
@@ -58,25 +61,54 @@ const InactivityMessage = () => {
     };
   }, []);
 
+  /* Starts a five minute timer when the inactivity window pops up. 
+     Logs the user out and refreshes the page if time elapses. */
+  useEffect(() => {
+    if (showPopup) {
+      logoutTimer.current = setInterval(() => {
+        if (seconds === 0 && minutes === 0) {
+          handleLogout();
+          window.location.reload(false);
+        } else if (seconds === 0) {
+          setSeconds(59);
+          setMinutes((minutes) => minutes - 1);
+        } 
+          else if (seconds > 0) {
+          setSeconds((seconds) => seconds - 1);
+        }
+      }, 1000)
+    }
+    return () => clearInterval(logoutTimer.current);
+  }, [showPopup, seconds, minutes])
```

Comment:
Thanks for the submission! Overall, this looks like a really useful improvement. I think we can improve the performance here though. Currently, this function will run every second. Can we make it more efficient? We really only need the function to run once, when the timer has expired.

We could use something like `setTimeout` to fire off the logout command after the time has expired. Then, if we exit early, use `clearTimeout` to suppress the logout. There may be a more 'react' way to do that as well.

Reply:
I could move the logout command into a setTimeout, but I'd still need setInterval to maintain the timer that's displayed to the user

Reply:
About the `setInterval`, I don't necessarily think we need the full timer to be running since this PR is meant to force logout for inactive users which might not care about the specific amount of time they have left on the application.

If we wish to keep a timer, I believe a minute count should suffice, for instance `5 minutes remaining`, `4 minutes remaining`, etc.

Reply:
@brancwill  That's a fair point. Looking at it again, how you're using `setInterval` makes sense. I guess my question then is why is the `useEffect` watching the `seconds` and `minutes`? It seems to me it can just fire off the `setInterval` on mount and then clean up after it.

I also don't think you need to keep seconds and minutes separate. You can just have one value like `const [secondsToLogout, setSecondsToLogout] = useState(300)` and then compute the minutes and seconds in the JSX.

```
return <div>{Math.floor(secondsToLogout/60)}:{secondsToLogout % 60}</div>
```

Reply:
If the seconds state wasn't in the dependency array, then the timer wouldn't update properly and subtract minutes like they were seconds. With the implementations you suggested however, that is no longer necessary.